### DISCUSSION TITLE
Execute Commands from Other Plugins

### DISCUSSION THREAD

**Discussion by macdrifter** [SilentVoid13/Templater#363] [Help]
Is there a syntax to execute a command from another Obsidian plugin as part of a template? 

Use case:

I would like a template that creates a new file after requesting a title and a URL. I would like to call the "Extract URL Content" plugin using the requested URL.

Current template looks like this:

```
<%*
let qcFileName = await tp.system.prompt("Note Title")
let qcURL = await tp.system.prompt("URL")
titleName = qcFileName + " " + tp.date.now("YYYY-MM-DD")
await tp.file.rename(titleName)
await tp.file.move("/Meta/Sources" + titleName);
-%>
---
title: <% qcFileName %>
date: <% tp.file.creation_date("YYYY-MM-DD HH:mm:ss") %>
link: <% qcURL %>
tags:
- source
---

```

I'd like to set the body below the YAML to be the extract URL results. This plugin can work with selected URLs or with URLs on the clipboard. I'm fine with switching up the logic to not request a URL and instead to just use the clipboard contents. I have been unable to figure out how to use Templater to get the results from another plugin and insert those results in the document.

**Comment by AB1908** [comment]
Ahhh I forgot how to do this. It was on Discord. Something by the name of `runCommandByID` or something similar.

**Comment by nabaco** [comment]
Any suggestions here?
I thought something like `<% tp.obsidian.app.plugins.getPlugin("plugin-name"). %>`  or maybe `<% tp.obisidian.command. %>`?

**Comment by willasm** [comment]
I use this in a dataviewjs block to call my user functions from templater, perhaps this could get you going in the right direction.

let tpl = this.app.plugins.plugins['templater-obsidian'].templater;
let wc = new tpl.current_functions_object.user.myUserFunction();

**Reply by nabaco** [reply]
yeah, I went in that direction: https://github.com/SilentVoid13/Templater/issues/937

**Comment by YukioE** [comment]
for anyone finding this in the future:

the javascript function is `app.commands.executeCommandById(ID);`

to get the ID of a command you need to open the dev tools console with `Ctrl+Shift+I`
and enter `app.commands.listCommands()` this will return an Array with all of the registered commands

![output](https://github.com/user-attachments/assets/87f2c098-3771-4763-a5f4-1409d55b883b)

the ID of a command is not a number, it is a string, e.g. `editor:save-file`

**Reply by makamto** [reply]
> for anyone finding this in the future:
> 
> the javascript function is `app.commands.executeCommandById(ID);`
> 
> to get the ID of a command you need to open the dev tools console with `Ctrl+Shift+I` and enter `app.commands.listCommands()` this will return an Array with all of the registered commands
> 
> ![output](https://private-user-images.githubusercontent.com/93155932/352823611-87f2c098-3771-4763-a5f4-1409d55b883b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE3OTk1NTEsIm5iZiI6MTc0MTc5OTI1MSwicGF0aCI6Ii85MzE1NTkzMi8zNTI4MjM2MTEtODdmMmMwOTgtMzc3MS00NzYzLWE1ZjQtMTQwOWQ1NWI4ODNiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzEyVDE3MDczMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY5NTY0NzVhZWE1OTU1NTI0Zjc1YTdhMjMyYzliMzE3MTFhMDkzODRjMDZjYjM4ZTc5MGI2YjY5N2JkYzRhYWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.InRTMA4mf6qUMlpdaKDZx0qz6fqdl2gkzVmw5w2x7ZM)
> 
> the ID of a command is not a number, it is a string, e.g. `editor:save-file`

Using `:obcommand` will also give you list of command too.
